### PR TITLE
gad-3 : workers would be assigned by Section Incharge instead of Caretaker

### PR DESCRIPTION
**Pull request by ujjawallele13**
## Proposed changes
1. Workers would be now managed by section incharge assigned for the type of work.
2. Section Incharge would assign the worker to the complaint.

### Brief description of what is fixed or changed
1. DB changes :- Section incharge would now be linked to the workers
2. UI changes :-  Add/Delete and assign/reassign of workers by the section incharge
     
## Types of changes

_Put an `x` in the boxes that apply_

-   [ ] Bugfix (non-breaking change which fixes an issue)
-   [x] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
-   [ ] Other (please describe):

## Checklist

_Put an `x` in the boxes that apply_

-   [x] My code follows the [style guidelines of this project](https://docs.google.com/document/d/1GI2Hile8UbGZ82gFe1y4wAVPcNPXip1cmXTzog1S41U/edit)
-   [x] I have performed a self-review of my own code
-   [x] I have created new branch for this pull request
-   [x] I have commented my code, particularly in hard-to-understand areas
-   [x] I have made corresponding changes to the documentation
-   [x] My changes generate no new warnings
-   [x] My changes does not break the current system and it passes all the current test cases.

## Screenshots

Please attach the screenshots of the changes made in case of change in user interface

Updated UI  for section incharge

secIncharge assign worker

secIncharge reassign worker

secIncharge manage workers

Below are the Updated UI  for caretaker

caretaker update status

caretaker mark as done

## Other information

DB changes :- 
1. Added a new user "electricity_incharge" with password "user@123"

incharge username

2. Added extra-info of "electricity_incharge" with id=2900 and user_type="staff"

incharge extra info

3. Added "electricity_incharge" in section Incharge table
 
electricity incharge entry

4. Added worker for the "electricity_incharge"

worker for incharge

### CODE REVIEW

**Review by PranshuNayak — COMMENTED** [review]
LGTM!

**Comment by PranshuNayak** [comment]
@Aksh-Bansal-dev this PR has some DB changes, can u please see that they are properly migrated after merge?

**Comment by Arnav511** [comment]
I am getting the following error

Screenshot (242)

Ping @ujjawallele13

**Comment by ujjawallele13** [comment]
> I am getting the following error
> 
> Screenshot (242)
> 
> Ping @ujjawallele13

You have to first migrate the changes which are done in the models of complaint_system @Arnav511

**Comment by Arnav511** [comment]
I have done that @ujjawallele13 
Screenshot (243)

**Comment by samay-rgb** [comment]
> I am getting the following error
> 
> Screenshot (242)
> 
> Ping @ujjawallele13



> I have done that @ujjawallele13 Screenshot (243)

You need to run migration commands whenever there is any db related change:
```
python  manage.py makemigrations
python manage.py migrate
```

File: FusionIIIT/applications/complaint_system/views.py [hunk 1]
```diff
@@ -26,64 +28,49 @@ def complaint_reassign(request,wid,iid):
...
-            message = "Your complaint has been redirected to another caretaker"
-            complaint_system_notif(request.user, complainer_details.complainer.user ,'comp_redirect_alert',complainer_details.id,student,message)
-            return HttpResponseRedirect(url)
         
     else:
         y = ExtraInfo.objects.all().select_related('user','department').get(id=y.id)
-        a = Caretaker.objects.select_related('staff_id','staff_id__user','staff_id__department').get(staff_id=y)
-        b = a.area
+        a = SectionIncharge.objects.select_related('staff_id','staff_id__user','staff_id__department').get(staff_id=y)
+        b = a.work_type
```
Comment (samay-rgb) [inline FusionIIIT/applications/complaint_system/views.py:44]: You can use better variable names instead of letters which will improve code readability.

**Comment by Arnav511** [comment]
I am still getting the same error @samay-rgb

**Comment by samay-rgb** [comment]
@Arnav511 Check if this works for you

1. Revert model.py changes 
2. Run the migration commands
3. Apply model.py changes
4. Run the migration commands again

**Comment by ujjawallele13** [comment]
Are there any issues @samay-rgb @PranshuNayak @Arnav511 @akshatnema

**Comment by Arnav511** [comment]
I am still getting the same error of relation doesn't exist @ujjawallele13 @samay-rgb

**Comment by PranshuNayak** [comment]
> Are there any issues @samay-rgb @PranshuNayak @Arnav511 @akshatnema

Your PR will be merged after some necessary changes have been completed in the docker file

**Comment by PranshuNayak** [comment]
> I am still getting the same error of relation doesn't exist @ujjawallele13 @samay-rgb

have you reverted back to the initial database schemas, the one you forked while setup? After you are done reverting follow the steps suggested by @samay-rgb

**Comment by Aksh-Bansal-dev** [comment]
@ujjawallele13 Can you pull the latest changes from the main branch? It will be easier to check the migrations with docker.

**Comment by akshatnema** [comment]
> @ujjawallele13 Can you pull the latest changes from the main branch? It will be easier to check the migrations with docker.

@ujjawallele13 Merge it with gad-3 only as I updated the upstream gad-3 branch with main.

**Comment by ujjawallele13** [comment]
Ok merged it. @Aksh-Bansal-dev @akshatnema

**Comment by akshatnema** [comment]
@Arnav511 now please review the PR as this PR now have the latest changes with docker files.

**Review by Aksh-Bansal-dev — APPROVED** [review]
Migrations are working fine.

**Comment by ujjawallele13** [comment]
Could this PR be merged now @Arnav511 @akshatnema @PranshuNayak @samay-rgb

**Review by samay-rgb — APPROVED** [review]

**Comment by PranshuNayak** [comment]
@akshatnema this pr can be merged

File: FusionIIIT/templates/complaintModule/complaint_secincharge_detail.html [hunk 2]
```diff
@@ -1,167 +1,167 @@
...
+            </a>
+
+            <a class="item" data-tab="third">
+                Unresolved Complaints
+                <i class="right floated chevron right icon"></i>
+            </a>
+            <a class="item" data-tab="fourth">
+                Overdue Complaints
+                <i class="right floated chevron right icon"></i>
+            </a>-->
```
Comment (akshatnema) [inline FusionIIIT/templates/complaintModule/complaint_secincharge_detail.html]:
```suggestion
           
```

Please remove the comments from the PR if not needed.

File: FusionIIIT/templates/complaintModule/complaint_secincharge_detail.html [hunk 3]
```diff
@@ -1,167 +1,167 @@
...
+
+            </div>
+
+
+            <div class="ui active tab large" data-tab="feedback">
+                <!--  <div class="field">
+                                  <button class="ui green icon button right floated">
+                                     <i class="large undo icon" onclick="back_to_caretaker_history(event)"></i>
+                                 </button>
+                         </div> -->
```
Comment (akshatnema) [inline FusionIIIT/templates/complaintModule/complaint_secincharge_detail.html]:
```suggestion
  
```

File: FusionIIIT/templates/complaintModule/complainthistory_secincharge.html [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+                        Type</b>
+                    </td>
+
+                    <td>
+                        <b>Location</b>
+                    </td>
+
+                   <!--  <th>
+                        Specific <br> Location
+                    </th> -->
```
Comment (akshatnema) [inline FusionIIIT/templates/complaintModule/complainthistory_secincharge.html]:
```suggestion
```

File: FusionIIIT/templates/complaintModule/complainthistory_secincharge.html [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+    </div>
+  
+
+
+
+
+
+
+    
+
```
Comment (akshatnema) [inline FusionIIIT/templates/complaintModule/complainthistory_secincharge.html]:
```suggestion
```

Don't add unnecessary spacing.

File: FusionIIIT/templates/complaintModule/resolvecomplaint_caretaker.html [hunk 6]
```diff
@@ -0,0 +1,131 @@
+{% load static %}
+{% block resolvecomplaint_caretaker %}
+
+    {% comment %}The tab menu starts here!{% endcomment %}
+    <div class="ui pointing secondary menu">
+        <!-- <a class="active item" data-tab="resolvedcomplaint">
+            Resolved Complaints
+        </a> -->
```
Comment (akshatnema) [inline FusionIIIT/templates/complaintModule/resolvecomplaint_caretaker.html]:
```suggestion
```

**Review by akshatnema — CHANGES_REQUESTED** [review]
Please remove the unnecessary comments from your changes.

**Comment by ujjawallele13** [comment]
Ok removed the unnecessary comments @akshatnema

**Review by akshatnema — APPROVED** [review]

**Pull request merged**